libs/layer-composer/src/generators/polygons/polygons.ts

```diff
@@ -0,0 +1,59 @@
+import type { LineLayerSpecification } from '@globalfishingwatch/maplibre-gl'
+import { Group } from '../../types'
+import { GeneratorType, MergedGeneratorConfig, PolygonsGeneratorConfig } from '../types'
+import { isConfigVisible } from '../utils'
+
+export const TRACK_HIGHLIGHT_SUFFIX = '_highlighted'
+
+const DEFAULT_COLOR = 'rgba(0, 193, 231, 1)'
+
+export type GlobalPolygonsConfig = MergedGeneratorConfig<PolygonsGeneratorConfig>
+
+class PolygonsGenerator {
+  type = GeneratorType.Polygons
+
+  _getStyleSources = (config: GlobalPolygonsConfig) => {
+    if (!config.data) {
```

Comment:
I also thought on using the geojson data as a [url](https://maplibre.org/maplibre-gl-js-docs/style-spec/sources/#geojson-data) so we don't need to handle the resource...
but then I didn't find a way to know where the resource was loading for the spinner of the sublayer

Reply:
https://github.com/GlobalFishingWatch/frontend/pull/1411